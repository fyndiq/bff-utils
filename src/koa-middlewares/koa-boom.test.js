const Boom = require('boom')
const KoaBoom = require('./koa-boom')

describe('KoaBoom Middleware', () => {
  const middleware = KoaBoom()

  it('should throw a ctx error if the error is a Boom Error', async () => {
    const ctx = { throw: jest.fn() }
    const next = () =>
      new Promise(() => {
        throw new Boom()
      })
    await middleware(ctx, next)
    expect(ctx.throw).toHaveBeenCalled()
    expect(ctx.throw.mock.calls[0]).toMatchSnapshot()
  })

  it('should re-throw other errors', async () => {
    const next = () =>
      new Promise(() => {
        throw new Error('Hey')
      })
    expect(middleware({}, next)).rejects.toMatchSnapshot()
  })
})
