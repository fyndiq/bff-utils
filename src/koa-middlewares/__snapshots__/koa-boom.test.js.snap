// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KoaBoom Middleware should re-throw other errors 1`] = `[Error: Hey]`;

exports[`KoaBoom Middleware should throw a ctx error if the error is a Boom Error 1`] = `
Array [
  500,
  "Internal Server Error",
]
`;
